core/sdk-core/src/test/java/software/amazon/awssdk/core/rules/IntegrationTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.rules;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.core.rules.testutil.TestDiscovery;
+import software.amazon.awssdk.protocols.jsoncore.JsonNode;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class IntegrationTest {
+    private static final TestDiscovery TEST_DISCOVERY = new TestDiscovery();
+
+    @ParameterizedTest
+    @MethodSource("validTestcases")
+    void checkValidRules(ValidationTestCase validationTestCase) {
+        EndpointRuleset ruleset = EndpointRuleset.fromNode(validationTestCase.contents());
+        List<ValidationError> errors = new ValidateUriScheme().visitRuleset(ruleset)
+                                                              .collect(Collectors.toList());
+        assertEquals(errors, Collections.emptyList());
+    }
+
+    @ParameterizedTest
+    @MethodSource("checkableTestCases")
+    void executeTestSuite(TestDiscovery.RulesTestcase testcase) {
+        testcase.testcase().execute(testcase.ruleset());
+    }
+
+    private Stream<ValidationTestCase> validTestcases() {
+        TestDiscovery discovery = new TestDiscovery();
```

Comment:
It's confusing to have two instances of TestDiscovery, is it possible to refactor?

Reply:
Oh good catch. Removed the local var.